infrastructure/tari_script/src/script.rs

```diff
@@ -637,11 +640,11 @@ impl TariScript {
     fn handle_to_ristretto_point(&self, stack: &mut ExecutionStack) -> Result<(), ScriptError> {
         let item = stack.pop().ok_or(ScriptError::StackUnderflow)?;
         let scalar = match &item {
-            StackItem::Hash(hash) => hash.as_slice(),
+            StackItem::Hash(hash) => hash.as_slice(), // TODO: confirm this isn't problematic
```

Comment:
I'm not sure what makes sense here. For context, is conversion from a 256-bit hash using `from_bytes_mod_order` problematic? I assume that hashes (Blake Sha3 etc.) have a good chance of not being in the canonical form? 

Perhaps we would need to provide a 512-bit hash type and either allow reduction to canonical scalar bytes and then conversion to a Ristretto point or only support scalar and 512-bit hash input for ToRistrettoPoint.

FWIW I can't find an instance of Hash to Ristretto point conversion being used in the Tari codebase.

Reply:
Separately, it was mentioned that the naming of the opcode may be confusing. Hashing to a group element is usually taken to mean something very different than what is done here, and confusion could have very bad results. Perhaps something like `ToPublicKey` or `ScalarToPublicKey` instead? I'm not sure the best naming for this.

Reply:
See #5818.